apps/api/src/app/notification-template/e2e/delete-notification-template.e2e.ts

```diff
@@ -139,6 +145,47 @@ describe('Delete notification template by id - /notification-templates/:template
 
     expect(response.body.message).to.contains('Could not find workflow with id');
   });
+
+  it('should delete the notification template along with the message templates', async function () {
+    const notificationTemplateService = new NotificationTemplateService(
+      session.user._id,
+      session.organization._id,
+      session.environment._id
+    );
+    const template = await notificationTemplateService.createTemplate();
+
+    const messageTemplateIds = template.steps.map((step) => step._templateId);
+
+    const messageTemplates = await messageTemplateRepository.find({
+      _environmentId: session.environment._id,
+      _id: { $in: messageTemplateIds },
+    });
+
+    expect(messageTemplates.length).to.equal(2);
+
+    await session.testAgent.delete(`/v1/notification-templates/${template._id}`).send();
+
+    const deletedNotificationTemplate = await notificationTemplateRepository.findOne({
+      _environmentId: session.environment._id,
+      _id: template._id,
+    });
+
+    expect(deletedNotificationTemplate).to.equal(null);
+
+    const deletedIntegration = (
+      await notificationTemplateRepository.findDeleted({ _environmentId: session.environment._id, _id: template._id })
+    )[0];
```

Comment:
```suggestion
    const [deletedIntegration] = await notificationTemplateRepository.findDeleted({ _environmentId: session.environment._id, _id: template._id });
```
Same vibes. 🙂 
Not to address it as it is stylistic and we have no defined preference for any.